File: src/main/java/org/telran/lecture_15_Popular_algo/practice/TwoPointers/Task04.js
// Дан отсортированный массив целых чисел и число target.
// Найти два числа в массиве, сумма которых равна target.

function sumTarget(numbers, target) {
    let left = 0;
    let right = numbers.length - 1;
    while (left !== right){
        let sum = numbers[left] + numbers[right];
        if (sum === target){
            return [numbers[left], numbers[right]]
        } else if(sum < target){
            left++;
        }else{
            right--;
        }
    }
    return [];
}

let numbers = [2, 4, 7, 11, 14, 16, 17, 19, 22]
let target = 400;

let result = sumTarget(numbers, target);
console.log(result);

File: src/main/java/org/telran/lecture_15_Popular_algo/practice/TwoPointers/Task04.java
package org.telran.lecture_15_Popular_algo.practice.TwoPointers;

// Дан отсортированный массив целых чисел и число target.
// Найти два числа в массиве, сумма которых равна target.

public class Task04 {

    public static void main(String[] args) {
        int[] arr = {1, 2, 3, 4, 5, 6, 7, 8, 9};
        int target = 10;

        findTwoSum(arr, target);
    }

    public static void findTwoSum(int[] arr, int target) {
        int left = 0;
        int right = arr.length - 1;

        while (left < right) {
            int sum = arr[left] + arr[right];

            if (sum == target) {
                System.out.println("Найдены числа: " + arr[left] + " и " + arr[right]);
                return;
            } else if (sum < target) {
                left++;
            } else {
                right--;
            }
        }

        System.out.println("Пара чисел с суммой " + target + " не найдена.");
    }
}
